tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/concurrent/ControlledWorkerScheduler.java

```diff
@@ -42,7 +43,8 @@ public class ControlledWorkerScheduler<T> implements Scheduler<T>, TaskWrapperAw
 
 	private final ExecutorService executor;
 
-	private final LinkedBlockingQueue<Runnable> _taskQueue = new LinkedBlockingQueue<>();
+	// Note: initialized in #createExecutorService
+	protected BlockingQueue<Runnable> _taskQueue;
```

Comment:
hmm, this is detected as API incompatibility. 

```
 Nov 11, 2024 1:43:39 PM japicmp.output.incompatible.IncompatibleErrorOutput warn
WARNING: Incompatibility detected: Requires semantic version level MAJOR: JApiField [changeStatus=MODIFIED, jApiClass=JApiClass [fullyQualifiedName=org.eclipse.rdf4j.federated.evaluation.concurrent.ControlledWorkerScheduler, changeStatus=MODIFIED, compatibilityChanges=[]], oldFieldOptional=org.eclipse.rdf4j.federated.evaluation.concurrent.ControlledWorkerScheduler._taskQueue:Ljava/util/concurrent/LinkedBlockingQueue;, newFieldOptional=org.eclipse.rdf4j.federated.evaluation.concurrent.ControlledWorkerScheduler._taskQueue:Ljava/util/concurrent/BlockingQueue;, annotations=[], accessModifier=japicmp.model.JApiModifier@7f5a7e9c, staticModifier=japicmp.model.JApiModifier@193c2d13, finalModifier=japicmp.model.JApiModifier@47c5d598, transientModifier=japicmp.model.JApiModifier@6dbebdeb, syntheticModifier=japicmp.model.JApiModifier@67c7088e, syntheticAttribute=japicmp.model.JApiAttribute@755aaf09, compatibilityChanges=[FIELD_TYPE_CHANGED], type=japicmp.model.JApiType@5ae9b65a]
```

I will rewrite it tomorrow morning with a getter in the constructor

Reply:
Hmm, it is possible to add this override rule to the main pom.xml

I would not do so however in this change, I am not sure what else is included in FIELD_TYPE_CHANGED other than removing the "final" modifier

```
<overrideCompatibilityChangeParameter>
	<compatibilityChange>FIELD_TYPE_CHANGED</compatibilityChange>
	<binaryCompatible>true</binaryCompatible>
	<sourceCompatible>true</sourceCompatible>
	<semanticVersionLevel>MINOR</semanticVersionLevel>
</overrideCompatibilityChangeParameter>
```

Reply:
It's changed from LinkedBlockingQueue to BlockingQueue. Which I assume isn't binary compatible, but should be source compatible since the field was private. 

Do you get the same issue if you leave it as private?